### PR TITLE
Revert remote url-related changes in #4652

### PR DESCRIPTION
**Pull request by annasong20**
Revert url changes and related changes in url-processing code paths in #4652 due to code freeze issue #4756.

### CODE REVIEW

File: api/internal/git/repospec.go [hunk 1]
```diff
@@ -214,29 +212,25 @@ func parseHostSpec(n string) (string, string) {
...
+			}
 		}
 		return host, n
 	}
 
+	// TODO(annasong): replace this array with Schemes
 	// If host is a http(s) or ssh URL, grab the domain part.
-	for _, p := range Schemes() {
+	for _, p := range []string{
+		"ssh://", "https://", "http://"} {
```
Comment (natasha41575) [inline api/internal/git/repospec.go:233]: This revert should include removing `Schemes` (L23) because it was introduced for this change

Comment (annasong20) [reply api/internal/git/repospec.go:233]: `Schemes` should be treated the same as `HasRemoteFileScheme` in fileloader.go. They are additions that don't affect the code path loading remotes.

Comment (natasha41575) [reply api/internal/git/repospec.go:233]: Are you using it anywhere else or planning to use it in a future PR?

I want to make sure we are not keeping around code that isn’t or won’t be used.

**Comment by natasha41575** [comment]
Discussed offline with @KnVerey. We agreed that it is easier to make sure we are doing the right thing by reverting the entire PR and then reintroducing the changes.

Reversion PR: https://github.com/kubernetes-sigs/kustomize/pull/4775

**Pull request closed without merging**